pkg/controller/volume/attachdetach/cache/actual_state_of_world_test.go

```diff
@@ -780,6 +794,42 @@ func Test_MarkDesireToDetach_Positive_UnsetWithInitialSetVolumeMountedByNodePres
 	verifyAttachedVolume(t, attachedVolumes, generatedVolumeName, string(volumeName), nodeName, false /* expectedMountedByNode */, true /* expectNonZeroDetachRequestedTime */)
 }
 
+// Populates data struct with one volume/node entry.
+// Calls SetDetachRequestTime twice and sleep maxWaitTime (1 second) in between
+// The elapsed time returned from the first SetDetachRequestTime call should be smaller than maxWaitTime
+// The elapsed time returned from the second SetDetachRequestTime call should be larger than maxWaitTime
+func Test_SetDetachRequestTime_Positive(t *testing.T) {
+	// Arrange
+	volumePluginMgr, _ := volumetesting.GetTestVolumePluginMgr(t)
+	asw := NewActualStateOfWorld(volumePluginMgr)
+	volumeName := api.UniqueVolumeName("volume-name")
+	volumeSpec := controllervolumetesting.GetTestVolumeSpec(string(volumeName), volumeName)
+	nodeName := "node-name"
+	devicePath := "fake/device/path"
+	generatedVolumeName, addErr := asw.AddVolumeNode(volumeSpec, nodeName, devicePath)
+	if addErr != nil {
+		t.Fatalf("AddVolumeNode failed. Expected: <no error> Actual: <%v>", addErr)
+	}
+
+	maxWaitTime := 1 * time.Second
+	etime, err := asw.SetDetachRequestTime(generatedVolumeName, nodeName)
+	if err != nil {
+		t.Fatalf("SetDetachRequestTime failed. Expected: <no error> Actual: <%v>", err)
+	}
+	if etime >= maxWaitTime {
+		t.Fatalf("SetDetachRequestTim Expected: <elapsed time %v is smaller than maxWaitTime %v> Actual <elapsed time is larger than maxWaitTime>", etime, maxWaitTime)
+	}
+	// Sleep and call SetDetachRequestTime again
+	time.Sleep(maxWaitTime)
```

Comment:
After the sleep, the check is below, to verify the etime should be equal to larger than maxWaitTime. There are two checks of the etime. The previous check is before sleep
if etime < maxWaitTime {
//fail
}